### PR TITLE
handle_vdev_rsc must return error if notifyid cannot be assigned

### PR DESCRIPTION
**Pull request by tammyleino**
Updated handle_vdev_rsc to return error if a unique id cannot be assigned
Signed-off-by: Tammy Leino <tammy_leino@mentor.com>

### CODE REVIEW

**Review by edmooring — APPROVED** [review]
Looks good to me.

File: lib/remoteproc/rsc_table_parser.c [hunk 1]
```diff
@@ -134,6 +134,8 @@ int handle_vdev_rsc(struct remoteproc *rproc, void *rsc)
 					  notifyid, notifyid + 1);
 	if (notifyid != RSC_NOTIFY_ID_ANY)
 		vdev_rsc->notifyid = notifyid;
+	else
+		return -RPROC_ERR_RSC_TAB_NP;
```
Comment (arnopo) [inline lib/remoteproc/rsc_table_parser.c:138]: reason to not apply this check also for `vring_rsc->notifyid`

File: lib/remoteproc/rsc_table_parser.c [hunk 2]
```diff
@@ -146,9 +150,21 @@ int handle_vdev_rsc(struct remoteproc *rproc, void *rsc)
...
 			vring_rsc->notifyid = notifyid;
+		else
+			goto err;
 	}
 
 	return 0;
+
+err:
+
+	for (i--; i >= 0; i--) {
```
Comment (arnopo) [inline lib/remoteproc/rsc_table_parser.c:161]: infinite loop?
warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
  161 |  for (i--; i >= 0; i--) {
      |              ^~

**Review by arnopo — CHANGES_REQUESTED** [review]

File: lib/remoteproc/rsc_table_parser.c (hunk 2 above)
Comment (tammyleino) [reply lib/remoteproc/rsc_table_parser.c:161]: Yes, sorry - I need to change i to int.

**Review by arnopo — APPROVED** [review]
a minor comment but LGTM

File: lib/remoteproc/rsc_table_parser.c [hunk 3]
```diff
@@ -146,9 +151,21 @@ int handle_vdev_rsc(struct remoteproc *rproc, void *rsc)
...
 		if (notifyid != RSC_NOTIFY_ID_ANY)
 			vring_rsc->notifyid = notifyid;
+		else
+			goto err;
 	}
 
 	return 0;
+
+err:
+
```
[the remark this answers is not in the document]
Comment (tammyleino) [reply lib/remoteproc/rsc_table_parser.c:161]: @arnopo Do you want lines 159 and 161 removed or just 161?

[the remark this answers is not in the document]
Comment (arnopo) [reply lib/remoteproc/rsc_table_parser.c:161]: just 161

**Review by edmooring — APPROVED** [review]
Looks good to go.

**Pull request merged**